pkg/install/install.go

```diff
@@ -425,12 +446,12 @@ func (i *Installer) deployARMTemplate(ctx context.Context, rg string, tName stri
 			},
 		})
 
-		if isDeploymentActiveError(err) {
+		if azureerrors.IsDeploymentActiveError(err) {
 			i.log.Printf("waiting for %s template to be deployed", tName)
 			err = i.deployments.Wait(ctx, rg, deploymentName)
 		}
 
-		if hasAuthorizationFailedError(err) {
+		if azureerrors.HasAuthorizationFailedError(err) {
```

Comment:
@nilsanderselde did you spot a reason why we couldn't remove the `wait.PollImmediateUntil` in deployARMTemplate lik you did in deployStorageTemplate?

Reply:
Ah, maybe I answered my own question - the difficulty is the interaction with the "Deployment failed." CloudError handling.